routing.go

```diff
@@ -112,15 +112,9 @@ func (r *routelet) watchRoutingInfo(ctx context.Context, env env, process string
 	var version *call.Version
 	for re := retry.Begin(); re.Continue(ctx); {
 		routingInfo, newVersion, err := env.GetRoutingInfo(ctx, process, version)
-		if err != nil {
-			// TODO(mwhittaker): Handle errors more gracefully.
-			r.env.SystemLogger().Error("cannot get routing info", err)
-			continue
-		}
-		version = newVersion
-		if err := r.update(routingInfo, version); err != nil {
+		if err != nil || r.update(routingInfo, newVersion) != nil {
```

Comment:
let's say ``err!=nil equals true`` if ``GetRoutingInfo`` succeeds but ``r.update`` fails, the first err will be ``nil`` (equals like false) and second one will be non-nil which means true and true || false will be true, we log the error

Reply:
You're not storing the error returned by `r.update()` anywhere.

Reply:
The return value of `r.update` is never assigned to `err` though, right? So it will log `err`, which has a value of `nil` (not the error from `r.update`).